app/controllers/cases/filters_controller.rb

```diff
@@ -133,6 +135,29 @@ def open
 
     private
 
+    def set_filtered_tab_counts
+      @global_nav_manager.current_page.tabs.each do |tab| 
+        tab.set_count(
+          filtered_count_for_tab(
+            cases_for_tab(tab)
+          )
+        )
+      end
```

Comment:
for this part,  could I suggest only to calculate the non-active tab,  as the active tab will be calculated in my_open anyway,  so it may just add the following line to filter them out 
again, this way to reduce the unnecessary query time when loading the page
`    def set_filtered_tab_counts
      @global_nav_manager.current_page.tabs.each do |tab| 
        if tab.name != @global_nav_manager.current_page_or_tab.name
          tab.set_count(filtered_count_for_tab(tab.cases))
        end
      end
    end`